apps/onlyoffice.sh

```diff
@@ -85,126 +99,30 @@ then
-
-    LogLevel warn
-    CustomLog ${APACHE_LOG_DIR}/access.log combined
-    ErrorLog ${APACHE_LOG_DIR}/error.log
-
-    # Just in case - see below
-    SSLProxyEngine On
-    SSLProxyVerify None
-    SSLProxyCheckPeerCN Off
-    SSLProxyCheckPeerName Off
-
-    # contra mixed content warnings
-    RequestHeader set X-Forwarded-Proto "https"
-
-    # basic proxy settings
-    ProxyRequests off
-
-    ProxyPassMatch (.*)(\/websocket)$ "ws://127.0.0.3:9090/$1$2"
-    ProxyPass / "http://127.0.0.3:9090/"
-    ProxyPassReverse / "http://127.0.0.3:9090/"
-
-    <Location />
-        ProxyPassReverse /
-    </Location>
-</VirtualHost>
-HTTPS_CREATE
-
-    if [ -f "$HTTPS_CONF" ];
+    if occ_command_no_check app:list --output=json | jq -e '.enabled | .documentserver_community' > /dev/null
     then
-        print_text_in_color "$IGreen" "$HTTPS_CONF was successfully created."
-        sleep 1
-    else
-        print_text_in_color "$IRed" "Unable to create vhost, exiting..."
-        print_text_in_color "$IRed" "Please report this issue here $ISSUES"
-        exit 1
+        occ_command app:remove documentserver_community
     fi
 fi
 
-# Install certbot (Let's Encrypt)
-install_certbot
+# Install OnlyOffice
+msg_box "We will now install Onlyoffice.
 
-# Generate certs, and auto-configure if successful
-if generate_cert "$SUBDOMAIN"
-then
-    # Generate DHparams chifer
-    if [ ! -f "$DHPARAMS_SUB" ]
-    then
-        openssl dhparam -dsaparam -out "$DHPARAMS_SUB" 4096
-    fi
-    printf "%b" "${IGreen}Certs are generated!\n${Color_Off}"
-    a2ensite "$SUBDOMAIN.conf"
-    restart_webserver
-    # Install OnlyOffice
-    occ_command app:install onlyoffice
-else
-    last_fail_tls $SCRIPTS/apps/onlyoffice.sh
-fi
+Please note that it might take very long time to install it, and you will not see any progress bar.
 
-# Set config for OnlyOffice
-if [ -d "$NC_APPS_PATH"/onlyoffice ]
+Please be paitent, don't abort."
+install_and_enable_app onlyoffice
+sleep 2
+if install_and_enable_app documentserver_community
 then
-    occ_command config:app:set onlyoffice DocumentServerUrl --value=https://"$SUBDOMAIN/"
     chown -R www-data:www-data "$NC_APPS_PATH"
-    occ_command config:system:set trusted_domains 3 --value="$SUBDOMAIN"
-# Add prune command
-    {
-    echo "#!/bin/bash"
-    echo "docker system prune -a --force"
-    echo "exit"
-    } > "$SCRIPTS/dockerprune.sh"
-    chmod a+x "$SCRIPTS/dockerprune.sh"
-    crontab -u root -l | { cat; echo "@weekly $SCRIPTS/dockerprune.sh"; } | crontab -u root -
-    print_text_in_color "$ICyan" "Docker automatic prune job added."
-    service docker restart
-    docker restart onlyoffice
-    print_text_in_color "$IGreen" "OnlyOffice is now successfully installed."
-    any_key "Press any key to continue... "
+    occ_command config:app:set onlyoffice DocumentServerUrl --value=https://"$(occ_command_no_check config:system:get overwrite.cli.url)/"
```

Comment:
that command isn't exaclty correct. it should be:
`occ_command config:app:set onlyoffice DocumentServerUrl --value="$(occ_command_no_check config:system:get overwrite.cli.url)apps/documentserver_community/"`

Reply:
Nice catch!